src/tojson.jl

```diff
@@ -72,69 +79,109 @@ function tojson(output::String, ts::ReportingTestSet)
         end
     end
 
-    # Flag set in walk!(), used for top-level status property
+    # Flag set in push_result!(), used for top-level status property
     any_failed = false
+    # All stdout from the top level test set, used for all tests.
     output = truncate_output(output)
 
     """
-        walk!(prefix, tests, testset)
+        push_result!(tests, result, name)
+
+    Push a test result called `name` to `tests`. Returns the pushed Dict or `nothing` if the test was not pushed.
+    """
+    function push_result!(tests, result::Test.Result, name)
+        status = nothing
+        message = nothing
+        test_code = nothing
+
+        if result isa Test.Pass
+            status = "pass"
+            test_code = "@test $(result.orig_expr)"
+        elseif result isa Test.Fail
+            status = "fail"
+            message = string(result)
+            test_code = "@test $(result.orig_expr)"
+        elseif result isa Test.LogTestFailure
+            status = "fail"
+            message = string(result)
+            test_code = "@test_logs $(join(result.patterns, ' ')) $(result.orig_expr)"
+        elseif result isa Test.Error
+            status = "error"
+            message = result.backtrace
+            test_code = "@test " * result.orig_expr
+        elseif result isa Test.Broken
+            if result.test_type === :skipped
+                return nothing
+            end
```

Comment:
I wonder if these should be factored out or moved to individual methods. But I'm not sure that really gains us much readability. (Thinking out loud, maybe you have thoughts as well, otherwise ignore this comment.)

Reply:
Okay I just saw #78 takes care of it.